core/services/blockhashstore/feeder.go

```diff
@@ -63,8 +71,34 @@ type Feeder struct {
 	errsLock      sync.Mutex
 }
 
+func (f *Feeder) startHeartbeats(ctx context.Context) {
+	timer := time.NewTimer(time.Duration(f.heartbeatPeriodTime) * time.Second)
+	for alive := true; alive; {
+		select {
+		case <-timer.C:
+			f.lggr.Infow("storing heartbeat blockhash",
+				"heartbeatPeriodTime (in seconds)", f.heartbeatPeriodTime,
+				"calling BHSContract::storeEarliest",
+			)
+			if err := f.bhs.StoreEarliest(ctx); err != nil {
+				f.lggr.Errorw("failed to store heartbeat blockhash",
+					"err", err,
+					"heartbeatPeriodTime (in seconds)", f.heartbeatPeriodTime,
+					"BHSContract::storeEarliest",
+				)
+			}
+		case <-ctx.Done():
+			alive = false
+			timer.Stop()
+			break
+		}
+	}
+}
+
 // Run the feeder.
 func (f *Feeder) Run(ctx context.Context) error {
+	go f.startHeartbeats(ctx)
```

Comment:
I don't think we want this here since `Run` itself is run periodically. We probably want to launch this heartbeat monitor prior to launching the regular feeder event loop.

Reply:
The `Start` method in the delegate seems like the right place.